merge back chore_release-8.3 into edge

TamarZanzouri opened the pull request:
merge back chore_release-8.3 into edge

emilyburghardt committed:
update 8.3.0 release notes for stable release (#17407)

<!--
Thanks for taking the time to open a Pull Request (PR)! Please make sure
you've read the "Opening Pull Requests" section of our Contributing
Guide:


https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

GitHub provides robust markdown to format your PR. Links, diagrams,
pictures, and videos along with text formatting make it possible to
create a rich and informative PR. For more information on GitHub
markdown, see:


https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

To ensure your code is reviewed quickly and thoroughly, please fill out
the sections below to the best of your ability!
-->

# Overview

<!--
Describe your PR at a high level. State acceptance criteria and how this
PR fits into other work. Link issues, PRs, and other relevant resources.
-->

Updated release notes for stable release.

## Test Plan and Hands on Testing

<!--
Describe your testing of the PR. Emphasize testing not reflected in the
code. Attach protocols, logs, screenshots and any other assets that
support your testing.
-->

No building on our end until the notes are live in the app, but
possibility of format breaking is very low.

## Changelog

<!--
List changes introduced by this PR considering future developers and the
end user. Give careful thought and clear documentation to breaking
changes.
-->

-removed specific mention of Evotips and Ultima PEEK pipettes
-changed overall message to reflect our API 2.22 versioning message:
"beta features for our commercial partners"

## Review requests

<!--
- What do you need from reviewers to feel confident this PR is ready to
merge?
- Ask questions.
-->

Just a double check on messaging to customers. 

## Risk assessment

<!--
- Indicate the level of attention this PR needs.
- Provide context to guide reviewers.
- Discuss trade-offs, coupling, and side effects.
- Look for the possibility, even if you think it's small, that your
change may affect some other part of the system.
- For instance, changing return tip behavior may also change the
behavior of labware calibration.
- How do your unit tests and on hands on testing mitigate this PR's
risks and the risk of future regressions?
- Especially in high risk PRs, explain how you know your testing is
enough.
-->

low.

---------

Co-authored-by: Josh McVey <yersh13@gmail.com>

sfoster1 committed:
fix(shared-data): tc lids are schema v2 now (#17427)

We need to not use any schema v3 labware in 8.3 because we're going to
change the schema. The thermocycler lids were only available in schema
3. Change that by making the lid v1 def a schema 2 def, and making a lid
v2 def that is a schema 3 def.

All the data is the same, and we even kept the stackLimit and added it
to labware schema 2, so all the machine behavior should be exactly the
same. The only other thing that's not in schema 2 now is the new lid
stuff support, but that code isn't in 8.3 so we should be good.

Closes EXEC-1180

## review
- seem ok?

## testing
- run a smoke test for these. the data is literally all the same and
nothing identity checks schemas, just whether data is there, and the
same data is all still there.

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: y3rsh <502770+y3rsh@users.noreply.github.com>

Laura-Danielle committed:
refactor(api): Move the plunger position starting position after an evotip seal command (#17438)

# Overview

Moving the plunger the position to 400ul rather than the maximum volume
of the 1000ul pipettes.

## Test Plan and Hands on Testing

- Verify that we can still push out liquid OK from the resin tips.

## Changelog

- Modify the plunger volume aspirated to a lower number to avoid large
pressure build-ups.

## Risk assessment

Low. Beta feature

sfoster1 committed:
fix(shared-data): evotips v1 are schema 2 now (#17428)

We had the evotips labware and adapter as schema 3 labware. This was
probably necessary while this was a 3-element stack, but now it is not,
and we need to avoid shipping schema 3 labware in 8.3 so that we retain
the ability to change it.

Closes EXEC-1179

## review
- seem fine?

## testing
- smoke test. nothing checks schema identity, just whether data is
there, and all the data that the code in 8.3 uses is still there.

mjhuff committed:
fix(app): fix multi-location tip selection during error recovery (#17454)

Closes RQA-3927

sanni-t committed:
docs(api): clarify in docs that Well.has_tip checks only for unused tips (#17455)

Cherry-picked commit for 8.3.0 release branch

Original PR: #17412 
Closes RQA-3790

# Overview

`Well.has_tip` property has been checking for only clean/ unused tips
since API v2.2 but the docstrings don't mention that. That has
understandably caused some confusion in protocol behaviors. This PR
helps mitigate that issue by clarifying the exact behavior of this
property

## Risk assessment

None.

Co-authored-by: Max Marrone <max@opentrons.com>

smb2268 committed:
chore(shared-data,app): Block evotips labware from PD, LL, and Quick Transfer (#17460)

smb2268 committed:
fix(app): translate close button text for ODD toasts (#17459)

fix RQA-3946

y3rsh committed:
chore(ci): update deprecated GitHub actions tools  (#17494)

TamarZanzouri committed:
fix(api): recover from stall and an overpressure when dropping tip (#17471)

TamarZanzouri committed:
merge conflicts

y3rsh commented on .github/workflows/pd-test-build-deploy.yaml:
We need the edge version of this one.

TamarZanzouri replied:
done

SyntaxColoring commented on shared-data/labware/definitions/3/opentrons_tough_pcr_auto_sealing_lid/2.json:
```diff
@@ -36,8 +36,8 @@
   "namespace": "opentrons",
-  "version": 1,
-  "schemaVersion": 3,
+  "version": 2,
+  "schemaVersion": 2,
```
This file, shared-data/labware/definitions/3/opentrons_tough_pcr_auto_sealing_lid/2.json, should have `"schemaVersion": "3"`.

(`"version": "2"` is correct, though.)

This looks like a correct merge, but there was a bug in `chore_release-8.3.0`. I think it's probably easiest if we quickly fix the bug in `chore_release-8.3.0`, then merge `chore_release-8.3.0` into this branch (`chore_mergeback`). I'll open that PR in a second.

SyntaxColoring replied:
https://github.com/Opentrons/opentrons/pull/17508

TamarZanzouri committed:
edge changes for pd build overflow

TamarZanzouri committed:
change labware def import from shared data

SyntaxColoring committed:
fix(shared-data): Fix schema version of opentrons_tough_pcr_auto_sealing_lid (#17508)

SyntaxColoring reviewed (approved):
LGTM once:

* `chore_release-8.3.0` is merged into `chore_mergeback` one more time to bring in #17508.
* CI is all passing

Remember when clicking the merge button to *not* do a squash merge, which is the default. We want a normal merge. GitHub may not allow you to do this through the web UI because of our repository settings, but it should let you do it on the command line, like:

```shell
git switch chore_mergeback
git pull
git switch edge
git pull
git merge chore_mergeback
git push
```

TamarZanzouri committed:
Merge branch 'chore_release-8.3.0' into chore_mergeback

TamarZanzouri committed:
get chore code changes and fixed import for LabwareDef

github-actions[bot] committed:
fix(analyses-snapshot-testing): heal chore_mergeback snapshots (#17510)

This PR was requested on the PR
https://github.com/Opentrons/opentrons/pull/17506

Co-authored-by: TamarZanzouri <5837799+TamarZanzouri@users.noreply.github.com>

y3rsh reviewed (approved):
# :shipit:

TamarZanzouri committed:
Merge branch 'edge' into chore_mergeback
